clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
I am not convinced why this must be JSON exclusive. Why not give users the flexibility to use other formats they want? Probably we should get some voice from the users instead of making a decision for them?